crates/joystick/src/windows.rs

```diff
@@ -81,88 +52,79 @@ impl Joystick {
         Joystick::default()
     }
 
-    pub fn name(&self) -> &str {
-        &self.name
-    }
-
     pub fn tap_a(&mut self) {
-        self.tap(KeyCode::BTN_EAST)
+        self.tap(XButtons::A)
     }
 
     pub fn tap_b(&mut self) {
-        self.tap(KeyCode::BTN_SOUTH)
+        self.tap(XButtons::B)
     }
 
     pub fn tap_x(&mut self) {
-        self.tap(KeyCode::BTN_NORTH)
+        self.tap(XButtons::X)
     }
 
     pub fn tap_y(&mut self) {
-        self.tap(KeyCode::BTN_WEST)
+        self.tap(XButtons::Y)
     }
 
     pub fn tap_lt(&mut self) {
-        self.tap(KeyCode::BTN_TL)
+        self.tap(XButtons::LB)
     }
 
     pub fn tap_rt(&mut self) {
-        self.tap(KeyCode::BTN_TR)
+        self.tap(XButtons::RB)
     }
 
+    // TODO(eein): We dont rear triggers yet
     pub fn tap_lt2(&mut self) {
-        self.tap(KeyCode::BTN_TL2)
+        // self.tap(XButtons::LB)
     }
 
     pub fn tap_rt2(&mut self) {
-        self.tap(KeyCode::BTN_TR2)
+        // self.tap(XButtons::RB)
     }
 
     pub fn tap_select(&mut self) {
-        self.tap(KeyCode::BTN_SELECT)
+        self.tap(XButtons::BACK)
     }
 
     pub fn tap_start(&mut self) {
-        self.tap(KeyCode::BTN_START)
+        self.tap(XButtons::START)
     }
 
     pub fn tap_dpad_up(&mut self) {
-        self.tap(KeyCode::BTN_DPAD_UP)
+        self.tap(XButtons::UP)
     }
 
     pub fn tap_dpad_down(&mut self) {
-        self.tap(KeyCode::BTN_DPAD_DOWN)
+        self.tap(XButtons::DOWN)
     }
 
     pub fn tap_dpad_left(&mut self) {
-        self.tap(KeyCode::BTN_DPAD_LEFT)
+        self.tap(XButtons::LEFT)
     }
 
     pub fn tap_dpad_right(&mut self) {
-        self.tap(KeyCode::BTN_DPAD_RIGHT)
+        self.tap(XButtons::RIGHT)
     }
 
     pub fn release_all(&mut self) {
-        let mut keys = vec![];
-        for key in &self.keys {
-            keys.push(InputEvent::new(EventType::KEY.0, key.code(), 0));
-        }
-        self.device.lock().unwrap().emit(&keys).unwrap();
+        self.gamepad.buttons = vigem_client::XButtons!();
     }
 
-    pub fn tap(&mut self, button: KeyCode) {
+    pub fn tap(&mut self, button: u16) {
```

Comment:
There's a bit of an issue here. This doesn't share the same API as the linux one, so we'll have issues if we want to use this in the TAS code. IMO there should probably be a `JoystickDev` trait or something to define a common interface for the two implementations.

Reply:
I suppose my intention is to actually have the raw `tap` as a private method and only expose the explicit functions - i'll remove the pub to make that clear